src/coreclr/vm/ceemain.cpp

```diff
@@ -947,6 +950,29 @@ void EEStartupHelper()
                                                 g_MiniMetaDataBuffMaxSize, MEM_COMMIT, PAGE_READWRITE);
 #endif // FEATURE_MINIMETADATA_IN_TRIAGEDUMPS
 
+#ifdef TARGET_UNIX
+        // Check if the current code is executing in the single file host or in libcoreclr.so. The libSystem.Native is linked
```

Comment:
Would it be better to just check `if (!g_coreclr_embedded)`?

Reply:
I was looking for a check like that and have failed to locate this one :-)

Reply:
I think you could also use the `CORECLR_EMBEDDED` define?

Reply:
Ah, nice. I have not noticed that the ceemain.cpp is one of the few files that are compiled separately for single file host and libcoreclr.so.

Reply:
@elinor-fung I have found that when S.P.C. is bundled, this `NativeLibrary::LoadLibraryByName` call doesn't work as the S.P.C. doesn't exist in the filesystem, just in the bundle. This causes e.g. the 
 Microsoft.NET.HostModel.Tests.BundleAndRun.TestWithRelativePaths` test to fail.
Do I need to pass something else than `SystemDomain::SystemAssembly()` to `NativeLibrary::LoadLibraryByName` to make it work?

Reply:
Ah, it seems I should pass in the DLLIMPORTSEARCHPATH_ASSEMBLYDIRECTORY.

Reply:
The problem was actually different, in the `EEStartupHelper`, the native dll search paths from the properties passed by the host are not set yet. Moving the check to the end of the `CreateAppDomainWithManager` resolved the issue.

Reply:
Sigh, yet one more host test failing.

Reply:
I think that might be a variation of https://github.com/dotnet/runtime/issues/75387? The case that is failing uses an extraction path that has a backslash on Unix and loading native libraries in `PAL_LoadLibraryDirect` goes through `FILEDosToUnixPathA`.

Reply:
Ah, right, running it locally with the `FILEDosToUnixPathA` modified to be dummy fixed the issue. There was one more mono related issue that I've fixed, but now it seems fine.